### ISSUE TITLE
Manually adding "AppData\Local\surveyjoin\surveyjoin\Cache" required?

### ISSUE THREAD

**Issue by James-Thorson-NOAA** [DFO-NOAA-Pacific/surveyjoin#47]
Hi all,

I'm excited to see the project progress, and thanks for the huge effort involved!

I'm trying it out for the first time, and was getting warnings below when running `cache_data()`, which went away when I manually added "\surveyjoin\surveyjoin\Cache" to my "AppData\Local\" directory.  Is there a way to add code to automatically make the intended directory, or perhaps I made a mistake in following the README directions?

![image](https://github.com/user-attachments/assets/aa45ee3c-a599-4bc9-8db3-05d5e0dfa954)

**Comment by seananderson** [comment]
The cache folder is determined by:

```r
rappdirs::user_cache_dir("surveyjoin")
```

And it's supposed to do something sensible that is OS specific.

Perhaps we need a `dir.create()` on that folder in some cases?

**Comment by ericward-noaa** [comment]
@James-Thorson-NOAA would you mind re-installing the package and trying again? Thanks!

**Comment by James-Thorson-NOAA** [comment]
Thanks for the quick response!  I removed `surveyjoin`, deleted `AppData\surveyjoin` manually, reinstalled `surveyjoin` and still get the warnings during `cache_data()` (see below).  This is on a NOAA machine, so perhaps there's some permissions issue (or some other ... but it's a fresh install of R 4.4.1 FWIW)

**Comment by seananderson** [comment]
The continuous integration R CMD check is [failing on Windows after it doesn't find the cached data](https://github.com/DFO-NOAA-Pacific/surveyjoin/actions/runs/10310764464/job/28543107910), so perhaps we just need to test this on Windows.

It looks like there was a `dir.create()` in there, so it wasn't just that.

**Comment by ericward-noaa** [comment]
@James-Thorson-NOAA I think this may be fixed. @Lewis-Barnett-NOAA did some debugging with me and I realized the paths are being handled differently on different OSs: https://www.rdocumentation.org/packages/rappdirs/versions/0.3.3/topics/user_cache_dir

I also made a few other changes, removing the dependency on taxize (which has been orphaned and we don't really need) and more error handling -- so that if the caching fails on CI, the build won't totally fail.

Anyway, give it a shot and report back if it still fails. Thanks!
